src/components/Pagination.tsx

```diff
@@ -1,78 +1,107 @@
+export type PageNumberAndSize = { pageNumber: number; pageSize: number };
+export type OffsetAndLimit = { offset: number; limit: number };
+
+// Use OffsetAndLimit as our default b/c that's what backend filters expect
+export const defaultPage: OffsetAndLimit = { offset: 0, limit: 100 };
 
 interface PaginationProps {
-  label: string;
+  page: readonly [PageNumberAndSize, Dispatch<PageNumberAndSize>] | readonly [OffsetAndLimit, Dispatch<OffsetAndLimit>];
   totalCount: number;
-  settings: PageSettings;
-  setSettings: Dispatch<SetStateAction<PageSettings>>;
 }
 
 export function Pagination(props: PaginationProps) {
-  const { label, settings, totalCount, setSettings } = props;
-  const { perPage, page } = settings;
+  const { totalCount } = props;
+  const [page, setPage] = props.page;
+  const { pageSize, pageNumber } = toPageNumberSize(page);
+
+  const hasPrevPage = pageNumber > 1;
+  const hasNextPage = pageNumber < totalCount / pageSize;
+  // Create the `1 - 100 of 1000` or `0 of 0`
+  const first = pageSize * (pageNumber - 1) + (totalCount ? 1 : 0);
+  const last = Math.min(pageSize * pageNumber, totalCount);
+  // Don't both with `0 - 0 of 0`
+  const showLast = totalCount > 0;
 
-  const hasNextPage = useMemo(() => page < totalCount / perPage, [page, perPage, totalCount]);
+  // Convert the new page back to whatever format the caller wants
+  function set(newPage: PageNumberAndSize): void {
+    if ("pageNumber" in props.page[0]) {
+      setPage(newPage as any);
+    } else {
+      setPage(toLimitAndOffset(newPage) as any);
+    }
+  }
 
   const tid = useTestIds(props, "pagination");
   return (
     <div css={Css.df.bGray300.bt.xs.gray500.px1.ml2.pt2.$} {...tid}>
-      <div css={Css.df.mya.mr2.$} {...tid.perPageLabel}>
-        {label} per page:
+      <div css={Css.df.mya.mr2.$} {...tid.pageSizeLabel}>
+        Page size:
       </div>
       <div css={Css.wPx(78).$}>
         <SelectField
           compact
-          {...tid.perPage}
+          label="Page Size"
           labelStyle="hidden"
-          label={`${label} per page`}
           options={pageOptions}
-          value={perPage}
-          onSelect={(val) => {
-            setSettings({ page: 1, perPage: val! });
-          }}
+          value={pageSize}
+          onSelect={(val) => set({ pageNumber: 1, pageSize: val! })}
+          {...tid.pageSize}
         />
       </div>
       <div css={Css.mla.mya.df.$}>
         <div css={Css.df.mya.mr2.$} {...tid.pageInfoLabel}>
-          {perPage * (page - 1) + 1} {hasNextPage ? `- ${perPage * page}` : ""} of {totalCount} {label}
+          {first} {showLast ? `- ${last}` : ""} of {totalCount}
         </div>
         <IconButton
-          {...tid.previousIcon}
           icon="chevronLeft"
-          color={Palette.LightBlue700}
-          onClick={() => setSettings({ page: settings.page - 1, perPage })}
-          disabled={settings.page === 1}
+          color={hasPrevPage ? Palette.LightBlue700 : Palette.Gray200}
```

Comment:
**thought:** Maybe IconButton should auto set the color to `Gray200` when the button is `disabled`.

Reply:
Ah yeah, agreed...it looks like it _would_ do that:

```
    <Icon icon={icon} color={color || (isDisabled ? Palette.Gray400 : iconColor)} inc={compact ? 2 : inc} />
```

But we skip that once `color` is set, which we're setting here b/c the default color is gray:

```
  const iconColor = contrast ? contrastIconColor : defaultIconColor;
...
   const defaultIconColor = Palette.Gray900;
```

Not sure why our default icon button color is gray 900? Seems like if its clickable, it should be blue?